generate-spec

```diff
@@ -712,15 +768,111 @@ if (count($schemas) == 0 && count($routes) == 0) {
 	Logger::error("app", "No spec generated");
 }
 
-if (count($openapi["paths"]) == 0) {
-	$openapi["paths"] = new stdClass();
-}
-
 ksort($schemas);
-$openapi["components"]["schemas"] = count($schemas) == 0 ? new stdClass() : $schemas;
 
 if ($useTags) {
 	$openapi["tags"] = $tags;
 }
 
-file_put_contents($out, json_encode($openapi, Helpers::jsonFlags()));
+$hasSingleScope = count($scopePaths) === 1;
+$fullScopePathArrays = [];
+
+if (!$hasSingleScope) {
+	$scopePaths['full'] = [];
+}
+
+foreach ($scopePaths as $scope => $paths) {
+	$openapiScope = $openapi;
+
+	if (count($paths) == 0) {
+		$paths = new stdClass();
+	}
+
+	$scopeSuffix = ($hasSingleScope || $scope === 'default') ? '' : '-' . $scope;
+	$openapiScope['info']['title'] .= $scopeSuffix;
+	$openapiScope['paths'] = $paths;
+
+	if ($scope !== 'full' && !$hasSingleScope) {
+		$fullScopePathArrays[] = $paths;
+	}
+
+	if ($scope === 'full') {
+		$openapiScope['paths'] = array_merge(...$fullScopePathArrays);
+		$openapiScope['components']['schemas'] = $schemas;
+	} else {
+		$usedSchemas = [];
+		foreach ($paths as $url => $urlRoutes) {
+			foreach ($urlRoutes as $httpMethod => $routeData) {
+				foreach ($routeData['responses'] as $statusCode => $responseData) {
+					if (empty($responseData['content'])) {
+						continue;
+					}
+
+					foreach ($responseData['content'] as $contentType => $contentData) {
+						if ($contentType === 'application/json') {
+							if (isset($contentData['schema']) && is_array($contentData['schema'])) {
+								$newSchemas = Helpers::collectUsedRefs($contentData['schema']);
+								$usedSchemas = array_merge($usedSchemas, $newSchemas);
+							}
+						} elseif (
+							(
+								in_array($contentType, ['*/*', 'text/css', 'application/octet-stream'], true)
+								|| str_starts_with($contentType, 'image/')
+							)
+							&& isset($contentData['schema']['type'], $contentData['schema']['format'])
+							&& $contentData['schema']['type'] === 'string'
+							&& $contentData['schema']['format'] === 'binary') {
+							Logger::debug("app", "Binary response from '$httpMethod $url' - Skipping schema reading");
+						} elseif (isset($contentData['schema'])) {
+							Logger::warning("app", "Could not read used schemas for response to '$httpMethod $url' with status code $statusCode with content type $contentType");
+						}
```

Comment:
Please remove this logic and just always try collecting refs for the response schema regardless of the content type or schema. We can not be sure that only certain responses contain refs and this code definitely doesn't catch all the possible types because some responses allow setting custom content types for example.